pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -38,28 +38,33 @@ def pattern_matching_optimization(
     r"""Quantum function transform to optimize a circuit given a list of patterns (templates).
 
     Args:
-        tape (QuantumTape): A quantum tape to be optimized.
+        tape (QNode or QuantumTape or Callable): A quantum circuit to be optimized.
         pattern_tapes(list(.QuantumTape)): List of quantum tapes that implement the identity.
         custom_quantum_cost (dict): Optional, quantum cost that overrides the default cost dictionary.
 
     Returns:
-        pennylane.QNode or qfunc or tuple[List[.QuantumTape], function]: If a QNode is passed,
-        it returns a QNode with the transform added to its transform program.
-        If a tape is passed, returns a tuple containing a list of
-        quantum tapes to be evaluated, and a function to be applied to these
-        tape executions.
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumTape], function]: The transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
 
     Raises:
         QuantumFunctionError: The pattern provided is not a valid QuantumTape or the pattern contains measurements or
             the pattern does not implement identity or the circuit has less qubits than the pattern.
 
     **Example**
 
-    First let's consider the following circuit where we want to replace sequence of two ``pennylane.S`` gates with a
-    ``pennylane.PauliZ`` gate.
+    >>> dev = qml.device('default.qubit', wires=5)
+
+    You can apply the transform directly on a :class:`QNode`, for that you need first to define a pattern that is to be
```

Comment:
Have seen this for the last few - it feels like the example has a slightly unnecessary thing tacked on. Is the motivation just to show the QNode use case?

Reply:
Yes exactly, applying on the QNode should be the preffered way of applying transforms